blockprod/src/detail/block_maker.rs

```diff
@@ -92,20 +93,70 @@ impl BlockMaker {
         Ok(returned_accumulator)
     }
 
-    pub fn make_block(
+    pub async fn make_block(
         &self,
         current_tip_id: Id<Block>,
-        accumulator: &dyn TransactionAccumulator,
+        transactions: &[SignedTransaction],
```

Comment:
Rule of thumb: If you pass a reference then have to clone it, then make the parameter by value. This way you give the option to the caller to clone it consume it.